### ISSUE TITLE
Validator "filled" not working for number inputs.

### ISSUE THREAD

**Issue by slepic** [nette/forms#228]
Version: 2.3.10 - latest (3.0.1)

### Bug Description
Inputs with type number are validated incorreclty when the input is required (using "filled" validator).
```
Nette.validators = {
		filled: function(elem, arg, val) {
			if (elem.type === 'number' && elem.validity.badInput) {
				return true;
			}
```

### Steps To Reproduce
```
$form->addText('name', 'label')
			->setType('number')
			->setRequired();
```

### Expected Behavior
The input element should not prevent submission of form when users inputs valid numbers in it.

### Possible Solution
Negate the second part of the condition:
```
if (elem.type === 'number' && !elem.validity.badInput) {
  return true;
}
```
Or flip the return value for that branch:
```
if (elem.type === 'number' && elem.validity.badInput) {
  return false;
}
```
Not sure which of these is the right way...

**Comment by dg** [comment]
I am unable to reproduce this problem (version 3.0.1)

**Comment by slepic** [comment]
Aha, I think I figured it out. I forgot to mention I was trying to put in a decimal number. Had no idea that number input won't accept that unless there is also a step attribute on that input with value being decimal number. When I added step="0.01" I was able to input numbers with 2 decimal places and it validated fine. So, its my bad.. sry for hassle...
I think a regex pattern will suit me better then...
Anyway, maybe it would be nice to have setters for this step/min/max atrributes for numeric inputs. Actualy may be good to implement a separate NumericControl which would offer this kind of setup, becaus it kinda does not make sense on non-numeric text inputs.
